core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -557,7 +559,10 @@ public PlanOptimizers(
                         ruleStats,
                         statsCalculator,
                         estimatedExchangesCostCalculator,
-                        ImmutableSet.of(new TransformFilteringSemiJoinToInnerJoin()))); // must run after PredicatePushDown
+                        ImmutableSet.of(
+                                new TransformFilteringSemiJoinToInnerJoin(),
+                                new InlineProjectIntoFilter(metadata),
+                                new SimplifyFilterPredicate(metadata)))); // must run after PredicatePushDown
```

Comment:
PPD might be meaningful also for the `InlineProjectIntoFilter` rule. However, that is a general purpose rule which should't create dependencies on other optimizations. I'll move the comment so that it applies to `TransformFilteringSemiJoinToInnerJoin` only.